From fb47478bac1ad686bc0e9a6fec36d7d82e42cbbb Mon Sep 17 00:00:00 2001
From: Dinh Nguyen <dinguyen@opensource.altera.com>
Date: Fri, 14 Aug 2015 16:57:34 -0500
Subject: [PATCH 165/172] FogBugz #317771: Fix all the memory leaks

unmap all of these locally scoped ioremap and of_iomap variables.

Signed-off-by: Dinh Nguyen <dinguyen@opensource.altera.com>
[czou:Original patch taken from
https://github.com/altera-opensource/linux-socfpga.git socfpga-4.1]
Signed-off-by: czou <cao.zou@windriver.com>
---
 arch/arm/mach-socfpga/ecc.c      | 8 +++-----
 arch/arm/mach-socfpga/l2_cache.c | 4 ++++
 arch/arm/mach-socfpga/ocram.c    | 2 ++
 3 files changed, 9 insertions(+), 5 deletions(-)

diff --git a/arch/arm/mach-socfpga/ecc.c b/arch/arm/mach-socfpga/ecc.c
index ed54215..c4d5208 100644
--- a/arch/arm/mach-socfpga/ecc.c
+++ b/arch/arm/mach-socfpga/ecc.c
@@ -130,16 +130,14 @@ int socfpga_init_a10_ecc(struct device_node *np, u32 intmask, int dual_port)
 
 	if (!sys_manager_base_addr) {
 		pr_err("SOCFPGA: sys-mgr is not initialized\n");
-		ret = -EINVAL;
-		goto out;
+		return -EINVAL;
 	}
 
 	ecc_ioaddr = of_iomap(np, 0);
 	if (!ecc_ioaddr) {
 		pr_err("ECC: Unable to find %s mapping in dtb\n",
 		       np->full_name);
-		ret = -EINVAL;
-		goto out;
+		return -EINVAL;
 	}
 
 	/* Disable ECC */
@@ -175,7 +173,7 @@ int socfpga_init_a10_ecc(struct device_node *np, u32 intmask, int dual_port)
 
 	/* Ensure data is written out */
 	wmb();
-
 out:
+	iounmap(ecc_ioaddr);
 	return ret;
 }
diff --git a/arch/arm/mach-socfpga/l2_cache.c b/arch/arm/mach-socfpga/l2_cache.c
index ff265e1..1d90ad5 100644
--- a/arch/arm/mach-socfpga/l2_cache.c
+++ b/arch/arm/mach-socfpga/l2_cache.c
@@ -46,6 +46,8 @@ void socfpga_init_arria10_l2_ecc(void)
 	writel(SOCFPGA_A10_MPU_CTRL_L2_ECC_EN, mapped_l2_edac_addr +
 	       SOCFPGA_A10_SYSMGR_L2_ECC_CTRL);
 
+	iounmap(mapped_l2_edac_addr);
+
 	pr_alert("SOCFPGA: Success Initializing L2 cache ECC for Arria10");
 
 	of_node_put(np);
@@ -72,6 +74,8 @@ void socfpga_init_l2_ecc(void)
 	/* Enable ECC */
 	writel(0x01, mapped_l2_edac_addr);
 
+	iounmap(mapped_l2_edac_addr);
+
 	pr_alert("SOCFPGA: Success Initializing L2 cache ECC");
 	of_node_put(np);
 
diff --git a/arch/arm/mach-socfpga/ocram.c b/arch/arm/mach-socfpga/ocram.c
index 3d77352..058230e 100644
--- a/arch/arm/mach-socfpga/ocram.c
+++ b/arch/arm/mach-socfpga/ocram.c
@@ -106,6 +106,8 @@ void socfpga_init_ocram_ecc(void)
 	writel(0x18, mapped_ocr_edac_addr);
 	writel(0x19, mapped_ocr_edac_addr);
 
+	iounmap(mapped_ocr_edac_addr);
+
 	pr_alert("SOCFPGA: Success Initializing OCRAM");
 
 	return;
-- 
1.9.1

